### DISCUSSION TITLE
New LivingDocs support

### DISCUSSION THREAD

**Discussion by Tiberriver256** [reqnroll/Reqnroll#37] [Ideas]
Since we're rebuilding this portion of SpecFlow from scratch, could we align our reporting with the rest of the Cucumber ecosystem?

If we made a plugin that exported test results as [Cucumber Messages](https://github.com/cucumber/messages/tree/main) it would make reqnroll compatible with any of the [other reporters](https://cucumber.io/docs/cucumber/reporting) out of the box.

I think this would save quite a bit of time vs. creating our own json schema, reporters, extensions for various CI/CD tools, etc.

**Comment by gasparnagy** [comment]
Wow @Tiberriver256! 🎯 That was exactly my plan!

Is your plugin open-source? Do you see a chance to integrate it to the main Reqnroll codebase (so that ppl don't have to add additional plugins for enabling reports)?

**Reply by Tiberriver256** [reply]
Ah, sorry to mislead Gaspar! I don't have anything built. Just some thoughts on the direction to head with building a new one.

Would love to help though and would love to contribute here for sure 👍

**Reply by gasparnagy** [reply]
You know... When you're eyes read what you want to read... 😄

I have an item on my todo list to describe what we need to do to achieve this. But probably it's better if I stop working for this week...

**Reply by mel-leo-liu** [reply]
Hi @gasparnagy, has anyone in the team working on the reporting capability for Reqnroll? e.g. building LivingDoc from scratch? Do you have a timeline for this piece of work? We are looking at migrating from SpecFlow to Reqnroll but unsure about the reporting bit.

**Reply by gasparnagy** [reply]
@pi-leo-liu The actual coding has not started yet, but we already started to think on the solution. With voluntary work I don't think it would be fair to set a deadline, but many people miss LivingDoc, so it is a top priority. If you can also contribute, you are absolutely welcome.

**Reply by clrudolphi** [reply]
> @pi-leo-liu The actual coding has not started yet, but we already started to think on the solution. With voluntary work I don't think it would be fair to set a deadline, but many people miss LivingDoc, so it is a top priority. If you can also contribute, you are absolutely welcome.

@gasparnagy How can I help? Is the source to LivingDoc available on GH?

**Reply by gasparnagy** [reply]
@clrudolphi No, the LivingDoc is a closed source product of Tricentis. At this moment, as the design is not fixed / documented yet, you cannot help yet in this, but please stay tuned, I try to make the first steps next week, so that you (or others) can help.

**Reply by mel-leo-liu** [reply]
@gasparnagy  Makes sense. I am more than happy to contribute. I am being backed up by the whole PI team here so please let us know what we could help.

**Reply by mel-leo-liu** [reply]
@gasparnagy maybe it is not relevant to this thread but have you tried out the allure adaptor for SpecFlow on Reqnroll? or is there a specific plan to implement an allure adaptor for Reqnroll?

**Reply by gasparnagy** [reply]
@pi-leo-liu I haven't tried, but it will not work out of the box. The Allure SpecFlow adapter was implemented by the Allure team. It's on my list to contact them to make a similar Reqnroll adapter. I'll try to ping them now.

**Reply by gasparnagy** [reply]
@pi-leo-liu Talked to the Allure team and they are working on the Reqnroll integration already, so hopefully we can also use that soon.

**Reply by mel-leo-liu** [reply]
This is great! Thanks @gasparnagy. Fingers crossed, we will have that available soon. Please let me know if anything that I could help on.

**Comment by gasparnagy** [comment]
FYI: ReportPortal got Reqnroll integration 💯! https://github.com/reportportal/reportportal/issues/2228

**Comment by gasparnagy** [comment]
FYI: https://reqnroll.net/news/2025/08/reqnroll-v3-0-released-with-reporting/
